cpp/inc/bond/protocol/detail/rapidjson_utils.h

```diff
@@ -22,6 +22,10 @@ bool try_lexical_convert(const char* str, uint16_t& result);
 } // namespace bond
 
 
-#ifdef BOND_HEADER_ONLY
+#ifdef BOND_LIB_TYPE
+#if BOND_LIB_TYPE == BOND_LIB_TYPE_HEADER
 #include "rapidjson_utils_impl.h"
 #endif
+#else
+#error BOND_LIB_TYPE is undefined
```

Comment:
Do we need this explicit error handling? Don't all our supported compilers emit an error if you redefine a macro or access one that isn't defined? Similar comment for `BOND_DETAIL_HEADER_ONLY_INLINE`.

Reply:
Redefinition is usually a warning. Accessing undefined one in pre-processor considers it as empty. We want to be sure that `BOND_LIB_TYPE` is explicitly defined. This will catch the cases where `config.h` is not included for example.

Reply:
OK. I thought they were always errors in both cases. Carry on.